pkg/subnet/public.go

```diff
@@ -0,0 +1,121 @@
+package subnet
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanche-cli/pkg/application"
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/key"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/ava-labs/avalanche-network-runner/utils"
+	"github.com/ava-labs/avalanchego/ids"
+	avago_constants "github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/formatting/address"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary"
+	"github.com/ava-labs/avalanchego/wallet/subnet/primary/common"
+)
+
+type PublicSubnetDeployer struct {
+	LocalSubnetDeployer
+	baseDir     string
+	privKeyPath string
+	network     models.Network
+	log         logging.Logger
+}
+
+func NewPublicSubnetDeployer(app *application.Avalanche, privKeyPath string, network models.Network) *PublicSubnetDeployer {
+	return &PublicSubnetDeployer{
+		LocalSubnetDeployer: *NewLocalSubnetDeployer(app),
+		baseDir:             app.GetBaseDir(),
+		privKeyPath:         privKeyPath,
+		log:                 app.Log,
+		network:             network,
+	}
+}
+
+func (d *PublicSubnetDeployer) Deploy(controlKeys []string, threshold uint32, chain, genesis string) (ids.ID, ids.ID, error) {
+	wallet, api, err := d.loadWallet()
+	if err != nil {
+		return ids.Empty, ids.Empty, err
+	}
+	vmID, err := utils.VMID(chain)
+	if err != nil {
+		return ids.Empty, ids.Empty, fmt.Errorf("failed to create VM ID from %s: %w", chain, err)
+	}
+
+	subnetID, err := d.createSubnetTx(controlKeys, threshold, wallet)
+	if err != nil {
+		return ids.Empty, ids.Empty, err
+	}
+	ux.Logger.PrintToUser(subnetID.String())
+
+	blockchainID, err := d.createBlockchainTx(chain, vmID, subnetID, []byte(genesis), wallet)
+	if err != nil {
+		return ids.Empty, ids.Empty, err
+	}
+	ux.Logger.PrintToUser("Endpoint for blockchain %q with VM ID %q: %s/ext/bc/%s/rpc", blockchainID.String(), vmID.String(), api, blockchainID.String())
+	return subnetID, blockchainID, nil
+}
+
+func (d *PublicSubnetDeployer) loadWallet() (primary.Wallet, string, error) {
+	ctx := context.Background()
+
+	var (
+		api       string
+		networkID uint32
+	)
+
+	switch d.network {
+	case models.Fuji:
+		api = constants.FujiAPIEndpoint
+		networkID = avago_constants.FujiID
+	case models.Mainnet:
+		api = constants.MainnetAPIEndpoint
+		networkID = avago_constants.MainnetID
+	default:
+		return nil, "", fmt.Errorf("unsupported public network")
+	}
+
+	sf, err := key.LoadSoft(networkID, d.privKeyPath)
+	if err != nil {
+		return nil, "", err
+	}
+
+	kc := sf.KeyChain()
+
+	wallet, err := primary.NewWalletFromURI(ctx, api, kc)
```

Comment:
I don't believe this will work for a ledger, you can't export the keychain the same way, but I'm not 1000% sure

Reply:
alright, we agreed to defer this until we actually implement ledger integration